climakitae/core/data_export.py

```diff
@@ -480,39 +485,88 @@ def _rchop(s, suffix):
 
 
 ## TMY export functions
-def _tmy_header(location_name, df):
+def _grab_dem_elev_m(lat, lon):
+    """
+    Pulls elevation value from the USGS Elevation Point Query Service,
+    lat lon must be in decimal degrees (which it is after cleaning)
+    Modified from:
+    https://gis.stackexchange.com/questions/338392/getting-elevation-for-multiple-lat-long-coordinates-in-python
+    """
+    url = r"https://epqs.nationalmap.gov/v1/json?"
+
+    # define rest query params
+    params = {"output": "json", "x": lon, "y": lat, "units": "Meters"}
+
+    # format query string and return value
+    result = requests.get((url + urllib.parse.urlencode(params)))
+    dem_elev_long = float(result.json()["value"])
```

Comment:
I got an error here in the last cell of the nb:

```
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
Cell In [32], line 3
      1 for sim, tmy in tmy_data_to_export.items(): 
      2     filename = 'TMY_{0}_{1}'.format(station_name.replace(" ", "_").replace("(","").replace(")",""), sim).lower()
----> 3     write_tmy_file(filename, tmy_data_to_export[sim], station_name, station_code, stn_lat, stn_lon, stn_state, file_ext="epw")

File ~/src/climakitae/climakitae/core/data_export.py:766, in write_tmy_file(filename_to_export, df, location_name, station_code, stn_lat, stn_lon, stn_state, file_ext)
    763 path_to_file = filename_to_export + ".epw"
    764 with open(path_to_file, "w") as f:
    765     f.writelines(
--> 766         _epw_header(location_name, station_code, state, timezone, df)
    767     )  # writes required header lines
    768     df_string = _epw_format_data(df).to_csv(sep=",", header=False, index=False)
    769     f.write(df_string)  # writes data in EPW format

File ~/src/climakitae/climakitae/core/data_export.py:569, in _epw_header(location_name, station_code, state, timezone, df)
    554 """
    555 Constructs the header for the TMY output file in .epw format
    556 Source: EnergyPlus Version 23.1.0 Documentation
    557 """
    559 # line 1 - location, location name, state, country, WMO, lat, lon
    560 # line 1 - location, location name, state, country, weather station number (2 cols), lat, lon, time zone, elevation
    561 line_1 = "LOCATION,{0},{1},USA,{2},{3},{4},{5},{6}\n".format(
    562     location_name.upper(),
    563     state,
    564     "Custom_{}".format(station_code),
    565     station_code,
    566     df["lat"].values[0],
    567     df["lon"].values[0],
    568     timezone,
--> 569     _grab_dem_elev_m(lat=df["lat"].values[0], lon=df["lon"].values[0]),
    570 )
    572 # line 2 - design conditions, leave blank for now
    573 line_2 = "DESIGN CONDITIONS\n"

File ~/src/climakitae/climakitae/core/data_export.py:502, in _grab_dem_elev_m(lat, lon)
    500 # format query string and return value
    501 result = requests.get((url + urllib.parse.urlencode(params)))
--> 502 dem_elev_long = float(result.json()["value"])
    503 # make sure to round off lat-lon values so they are not improbably precise for our needs
    504 dem_elev_short = np.round(dem_elev_long, decimals=2)

KeyError: 'value'
```

Reply:
I then reran the cell and got an answer. I think it is a URL failure to retrieve from the API. Maybe we should add some error checking here.